packages/astro/src/@types/astro.ts

```diff
@@ -131,6 +131,18 @@ export interface AstroGlobal extends AstroGlobalPartial {
 	 * [Astro reference](https://docs.astro.build/en/reference/api-reference/#astrorequest)
 	 */
 	request: Request;
+	/** Information about the outgoing response. This is a standard [ResponseInit](https://developer.mozilla.org/en-US/docs/Web/API/Response/Response#init) object
+	 * 
+	 * For example, to change the status code you can set a different status on this object:
+	 * ```typescript
+	 * Astro.response.status = 404;
+	 * ```
+	 * 
+	 * [Astro reference](https://docs.astro.build/en/reference/api-reference/#astroresponse)
+	 */
+	response: ResponseInit & {
+		headers: Headers
```

Comment:
nit: mark this as `readonly` so users get build errors from TS if they try to reassign `Astro.response.headers`